# **ESTUDO DE CASO: ANÁLISE EXPLORATÓRIA DE DADOS**
#### **Case**: *Papelaria*
#### **Descrição**: *A área de analytics de uma empresa de varejo em papelaria precisa responder uma série de questões de negócio relacionadas às vendas do último mês. Os resultados ajudarão a empresa a priorizar o estoque e reposição dos itens mais vendidos (ou mais rentáveis); otimizar o mix de produtos oferecidos na loja; e tomar decisões de precificação e negociação com fornecedores.*
#### **Lista de Variáveis**:
- ##### *COD_PRODUTO: Código de identificação do produto*
- ##### *CATEGORIA_PRODUTO: Categoria do produto*
- ##### *SECAO_PRODUTO: Seção/departamento do produto*
- ##### *PRECO_UNITARIO: Preço unitário de venda do produto, vigente no mês anterior*
- ##### *QTD_UNIDADES_VENDIDAS_MES: Quantidade de unidades vendidas do produto no último mês*

#### **Base de Dados**: *Papelaria.txt*
--------------

#### **Tarefa**: Responda as 8 questões a seguir, utilizando as técnicas de análise exploratória de dados e visualização de dados mais pertinentes em cada situação. Para todas as questões, descreva os resultados obtidos e escreva suas conclusões. Cada questão vale 1,25 pontos.

#### **Entregável**: Este mesmo notebook do Python (extensão .ipynb) já compilado, com seus códigos em Python e conclusões. Não altere a estrutura geral do notebook.

#### **Data máxima de entrega**: 24/06/2025

------------


### **CARREGAMENTO DE BIBLIOTECAS**

In [1]:
import pandas as pd
import numpy as np

### **LEITURA DA BASE DE DADOS**

In [2]:
database = pd.read_table(filepath_or_buffer='Papelaria.txt',
                                     sep='\t',
                                     decimal='.',
                                     header=0)

In [3]:
#validar se existe dados duplicados
database['COD_PRODUTO'].duplicated().sum()

np.int64(0)

In [4]:
#validar se existe dados com valor ausente
database.isna().sum()

,0
COD_PRODUTO,0
CATEGORIA_PRODUTO,0
SECAO_PRODUTO,0
PRECO_UNITARIO,0
QTD_UNIDADES_VENDIDAS_MES,0


### **QUESTÃO 1**

#### Quais categorias de produto possuem produtos maior preço unitário, em média?

In [5]:
#Observar os valores CATEGORIA_PRODUTO e SECAO_PRODUTO agrupados com o PRECO_UNITARIO médio, apresentando os 15 maiores resultados.
database_questao1 = database.groupby(['CATEGORIA_PRODUTO','SECAO_PRODUTO'])['PRECO_UNITARIO'].mean().round(2)
database_questao1.nlargest(15)

,,PRECO_UNITARIO
CATEGORIA_PRODUTO,SECAO_PRODUTO,
Notebook Premium,Informática,5216.63
Notebook,Informática,3779.90
Tablet,Informática,2919.70
Impressora a Laser,Informática,1479.95
Monitor,Informática,887.15
Impressora a Jato,Informática,533.90
HD Externo,Informática,296.70
Teclado Mecânico,Informática,275.90
Mochila Executiva,Acessórios,254.92


In [6]:
#Observar os produtos ordenados por preco unitario decrescente, para avaliar se existe algum produto com valor muito alto
database.sort_values(by=['PRECO_UNITARIO'],ascending=[False]).head(10)

,COD_PRODUTO,CATEGORIA_PRODUTO,SECAO_PRODUTO,PRECO_UNITARIO,QTD_UNIDADES_VENDIDAS_MES
241,COD_P_255913,Notebook Premium,Informática,5850.0,1
144,COD_P_316388,Notebook Premium,Informática,4999.9,0
179,COD_P_146931,Notebook Premium,Informática,4800.0,1
278,COD_P_279132,Tablet,Informática,4399.9,4
98,COD_P_211988,Notebook,Informática,4399.9,0
284,COD_P_230700,Notebook,Informática,4299.9,1
74,COD_P_220538,Notebook,Informática,3699.9,1
11,COD_P_280958,Notebook,Informática,3299.9,2
249,COD_P_230978,Notebook,Informática,3199.9,2
54,COD_P_263117,Tablet,Informática,2980.0,6


In [7]:
#Observar somente a categoria 'Notebook Premium' ordenados por preco unitario decrescente
database[database['CATEGORIA_PRODUTO'] == 'Notebook Premium'].sort_values(by=['PRECO_UNITARIO'],ascending=[False])

,COD_PRODUTO,CATEGORIA_PRODUTO,SECAO_PRODUTO,PRECO_UNITARIO,QTD_UNIDADES_VENDIDAS_MES
241,COD_P_255913,Notebook Premium,Informática,5850.0,1
144,COD_P_316388,Notebook Premium,Informática,4999.9,0
179,COD_P_146931,Notebook Premium,Informática,4800.0,1


Observando os resultados da média de preço unitário por categoria, as categorias de produto que apresentam os maiores valores médios são: Notebook Premium em primeiro lugar, seguido por Notebook, Tablet, Impressora a Laser, Monitor, Impressora a Jato, HD Externo, Teclado Mecânico e Mochila Executiva.

Este padrão é notável, pois a categoria "Notebook Premium" alcança uma média de R\$ 5216.63, impulsionada por ter apenas 3 produtos, com um deles (COD_P_255913) superando R\$ 5.000,00 e, consequentemente, elevando a média. Em seguida, a categoria "Notebook" aparece na segunda posição, com uma diferença de aproximadamente R$ 1.500,00 em relação ao "Notebook Premium". Os quatro primeiros resultados ('Notebook Premium', 'Notebook', 'Tablet', 'Impressora a Laser') demonstram uma disparidade de preços significativa quando comparados às demais categorias, que posteriormente começam a apresentar valores mais próximos. Um detalhe importante é que os 8 produtos mais caros da base pertencem à Seção de produtos 'Informática'.

### **QUESTÃO 2**

#### Dentro de cada seção, quais categorias de produtos mais contribuíram com o faturamento bruto (produto entre a quantidade de unidades vendidas e o preço unitário)?

In [8]:
# Criação de base e dos campos com valor do faturamento bruto e %
database_questao2 = database.copy()
database_questao2['FATURAMENTO_BRUTO'] = database_questao2['PRECO_UNITARIO'] * database_questao2['QTD_UNIDADES_VENDIDAS_MES']
database_questao2['FATURAMENTO_BRUTO%'] = (
        (database_questao2['PRECO_UNITARIO'] * database_questao2['QTD_UNIDADES_VENDIDAS_MES']
         ) / database_questao2['FATURAMENTO_BRUTO'].sum().round(3)) * 100


In [9]:
# Mostrar os principais produtos agrupados por seção e categorias.
database_questao2_agrup = database_questao2.groupby(['SECAO_PRODUTO', 'CATEGORIA_PRODUTO']).agg(
        PRECO_UNITARIO_MEDIA=('PRECO_UNITARIO', 'mean'),
        QTD_UNIDADES_VENDIDAS_MES_SUM=('QTD_UNIDADES_VENDIDAS_MES', 'sum'),
        FATURAMENTO_BRUTO_SUM=('FATURAMENTO_BRUTO', 'sum'),
        FATURAMENTO_BRUTO_PERCENT_SUM=('FATURAMENTO_BRUTO%', 'sum')
).round(2).sort_values(by=["SECAO_PRODUTO", "FATURAMENTO_BRUTO_SUM"], ascending=[True, False])

# Mostrar os 5 maiores de cada seção pela ordenaçao do faturamento
database_questao2_agrup = database_questao2_agrup.groupby('SECAO_PRODUTO').head(5)
database_questao2_agrup



PRECO_UNITARIO_MEDIA  \
SECAO_PRODUTO CATEGORIA_PRODUTO                              
Acessórios    Mochila Escolar                       141.35   
              Fone de Ouvido sem Fio                130.28   
              Fone de Ouvido com Fio                 61.05   
              Mochila Executiva                     254.92   
              Caixa de Som Bluetooth                199.90   
Informática   Tablet                               2919.70   
              Notebook                             3779.90   
              Monitor                               887.15   
              Cartucho de Tinta                     104.90   
              Impressora a Jato                     533.90   
Papelaria     Lápis de Cor e Giz                     37.02   
              Caneta Especial                        34.83   
              Agenda                                 42.63   
              Caderno 10 Matérias                    20.34   
              Pasta                                  11.23   

                                      QTD_UNIDADES_VENDIDAS_MES_SUM  \
SECAO_PRODUTO CATEGORIA_PRODUTO                                       
Acessórios    Mochila Escolar                                    45   
              Fone de Ouvido sem Fio                             40   
              Fone de Ouvido com Fio                             65   
              Mochila Executiva                                  12   
              Caixa de Som Bluetooth                             14   
Informática   Tablet                                             17   
              Notebook                                            6   
              Monitor                                            19   
              Cartucho de Tinta                                 140   
              Impressora a Jato                                  22   
Papelaria     Lápis de Cor e Giz                                169   
              Caneta Especial                                   130   
              Agenda                                             46   
              Caderno 10 Matérias                                75   
              Pasta                                             146   

                                      FATURAMENTO_BRUTO_SUM  \
SECAO_PRODUTO CATEGORIA_PRODUTO                               
Acessórios    Mochila Escolar                        6203.2   
              Fone de Ouvido sem Fio                 4274.0   
              Fone de Ouvido com Fio                 4076.0   
              Mochila Executiva                      3058.9   
              Caixa de Som Bluetooth                 2118.6   
Informática   Tablet                                50775.3   
              Notebook                              20999.4   
              Monitor                               16444.8   
              Cartucho de Tinta                     11671.0   
              Impressora a Jato                     11507.8   
Papelaria     Lápis de Cor e Giz                     5416.1   
              Caneta Especial                        2696.0   
              Agenda                                 1927.0   
              Caderno 10 Matérias                    1458.1   
              Pasta                                  1216.8   

                                      FATURAMENTO_BRUTO_PERCENT_SUM  
SECAO_PRODUTO CATEGORIA_PRODUTO                                      
Acessórios    Mochila Escolar                                  3.01  
              Fone de Ouvido sem Fio                           2.07  
              Fone de Ouvido com Fio                           1.98  
              Mochila Executiva                                1.48  
              Caixa de Som Bluetooth                           1.03  
Informática   Tablet                                          24.61  
              Notebook                                        10.18  
              Monitor                                          7.97  
   

In [10]:
# Valor do faturamento por seção
database_questao2.groupby(['SECAO_PRODUTO'])[['FATURAMENTO_BRUTO','FATURAMENTO_BRUTO%']].sum().round(2)

,FATURAMENTO_BRUTO,FATURAMENTO_BRUTO%
SECAO_PRODUTO,,
Acessórios,25684.70,12.45
Informática,160444.40,77.77
Papelaria,20189.91,9.79


A análise do faturamento bruto revela que as contribuições variam significativamente entre as seções de produtos:
* A seção Papelaria tem o menor impacto no faturamento total, respondendo por 9.79%. Suas 5 principais categorias contribuidoras são 'Lápis de Cor e Giz', 'Caneta Especial', 'Agenda', 'Caderno 10 Matérias' e 'Pasta'. Observa-se que, apesar de contribuir menos para o faturamento total, as categorias nesta seção frequentemente apresentam um volume maior de unidades vendidas, compensando seus preços unitários médios mais baixos.
* A seção de Acessórios contribui com 12.45% do faturamento bruto total. As 5 principais categorias nesta seção são 'Mochila Escolar', 'Fone de Ouvido sem Fio', 'Fone de Ouvido com Fio', 'Mochila Executiva' e 'Caixa de Som Bluetooth'.
* A seção Informática detém a maior fatia do faturamento bruto, com 77.77% do total. As 5 categorias que mais contribuíram para esta seção são 'Tablet', 'Notebook', 'Monitor', 'Cartucho de Tinta' e 'Impressora a Jato'. Destaca-se que 'Tablet' e 'Notebook' são categorias com participação significativa no faturamento total, representando quase 25% e 10% do faturamento geral da loja, respectivamente. Este alto impacto ocorre mesmo sem um alto volume de unidades vendidas, sendo impulsionado principalmente pelo elevado preço médio desses produtos.


### **QUESTÃO 3**

#### Existe relação entre o preço unitário e a quantidade de unidades vendidas? Essa relação muda a depender da seção do produto?

In [11]:
database_questao3 = database.copy()

# Selecionando as variáveis
vendas =  database_questao3['QTD_UNIDADES_VENDIDAS_MES']
preco = database_questao3['PRECO_UNITARIO']

# Montando os cortes, escolhendo os quartis

cortes = [
    0,
    0.01,
    vendas.quantile(0.25),
    vendas.quantile(0.50) ,
    vendas.quantile(0.75),
    vendas.max()]

labels_qtd = ['Nenhum', 'Baixa', 'Média', 'Alta', 'Muito Alta']

database_questao3['QTD_UNIDADES_VENDIDAS_MES_FAIXAS'] = pd.cut(
    vendas,
    labels=labels_qtd,
    bins=cortes,
    include_lowest=True)

cortes = [
    0,
    preco.quantile(0.25),
    preco.quantile(0.50),
    preco.quantile(0.75),
    preco.max()]

labels_preco = ['Muito Barato','Barato','Preço Médio','Caro']

database_questao3['PRECO_UNITARIO_FAIXAS']  = pd.cut(
    preco,
    bins=cortes,
    labels=labels_preco,
    include_lowest=True)

database_questao3['PRECO_UNITARIO_FAIXAS'].value_counts().sort_index()

,count
PRECO_UNITARIO_FAIXAS,
Muito Barato,73
Barato,72
Preço Médio,77
Caro,68


In [12]:
database_questao3

,COD_PRODUTO,CATEGORIA_PRODUTO,SECAO_PRODUTO,PRECO_UNITARIO,QTD_UNIDADES_VENDIDAS_MES,QTD_UNIDADES_VENDIDAS_MES_FAIXAS,PRECO_UNITARIO_FAIXAS
0,COD_P_233209,Tinta e Pincel,Papelaria,5.9,14,Alta,Barato
1,COD_P_235103,Caneta Especial,Papelaria,5.9,26,Muito Alta,Barato
2,COD_P_178413,Monitor,Informática,799.9,3,Baixa,Caro
3,COD_P_316368,Mochila Escolar,Acessórios,139.0,3,Baixa,Caro
4,COD_P_190632,Suporte de Celular,Acessórios,29.9,5,Média,Preço Médio
...,...,...,...,...,...,...,...
285,COD_P_258390,Fone de Ouvido com Fio,Acessórios,49.9,2,Baixa,Preço Médio
286,COD_P_171592,Caderno 1 Matéria,Papelaria,9.9,15,Alta,Barato
287,COD_P_179866,Fone de Ouvido com Fio,Acessórios,39.8,5,Média,Preço Médio
288,COD_P_239935,Caderno 1 Matéria,Papelaria,2.9,9,Alta,Muito Barato


In [13]:
# cruzamento entre as 2 variaveis, criando uma tabela de frequência
pd.crosstab(
    database_questao3['QTD_UNIDADES_VENDIDAS_MES_FAIXAS'],
    database_questao3['PRECO_UNITARIO_FAIXAS'],
    normalize = 'columns',
    margins=True,
    margins_name='Total').round(2) * 100

PRECO_UNITARIO_FAIXAS,Muito Barato,Barato,Preço Médio,Caro,Total
QTD_UNIDADES_VENDIDAS_MES_FAIXAS,,,,,
Nenhum,0.0,6.0,5.0,6.0,4.0
Baixa,7.0,24.0,26.0,53.0,27.0
Média,11.0,17.0,27.0,21.0,19.0
Alta,32.0,35.0,19.0,16.0,26.0
Muito Alta,51.0,19.0,22.0,4.0,24.0


In [14]:
#cruzamento entre as 2 variaveis, agrupando por secao
pd.crosstab(
    index=[database_questao3['SECAO_PRODUTO'], database_questao3['QTD_UNIDADES_VENDIDAS_MES_FAIXAS']],
    columns=database_questao3['PRECO_UNITARIO_FAIXAS'],
    normalize = 'columns',
    margins=True,
    margins_name='Total').round(2) * 100

PRECO_UNITARIO_FAIXAS                           Muito Barato  Barato  \
SECAO_PRODUTO QTD_UNIDADES_VENDIDAS_MES_FAIXAS                         
Acessórios    Nenhum                                     0.0     0.0   
              Baixa                                      0.0     3.0   
              Média                                      0.0     1.0   
              Alta                                       0.0     3.0   
              Muito Alta                                 0.0     0.0   
Informática   Nenhum                                     0.0     0.0   
              Baixa                                      0.0     0.0   
              Média                                      0.0     0.0   
              Alta                                       0.0     0.0   
              Muito Alta                                 0.0     0.0   
Papelaria     Nenhum                                     0.0     6.0   
              Baixa                                      7.0    21.0   
              Média                                     11.0    15.0   
              Alta                                      32.0    32.0   
              Muito Alta                                51.0    19.0   

PRECO_UNITARIO_FAIXAS                           Preço Médio  Caro  Total  
SECAO_PRODUTO QTD_UNIDADES_VENDIDAS_MES_FAIXAS                            
Acessórios    Nenhum                                    4.0   1.0    1.0  
              Baixa                                    10.0  22.0    9.0  
              Média                                     9.0   9.0    5.0  
              Alta                                      4.0   7.0    3.0  
              Muito Alta                                4.0   0.0    1.0  
Informática   Nenhum                                    0.0   4.0    1.0  
              Baixa                                     3.0  29.0    8.0  
              Média                                     1.0  10.0    3.0  
              Alta                                      5.0   9.0    3.0  
              Muito Alta                               14.0   4.0    5.0  
Papelaria     Nenhum                                    1.0   0.0    2.0  
              Baixa                                    13.0   1.0   11.0  
              Média                                    17.0   1.0   11.0  
              Alta                                     10.0   0.0   19.0  
              Muito Alta                                4.0   0.0   19.0

In [15]:
# Isolando a seção "Acessórios"
database_questao3_filter = database_questao3[database_questao3['SECAO_PRODUTO'] == 'Acessórios']
pd.crosstab(database_questao3_filter['QTD_UNIDADES_VENDIDAS_MES_FAIXAS'], database_questao3_filter['PRECO_UNITARIO_FAIXAS'], normalize = 'columns', margins=True, margins_name='Total').round(2) * 100

PRECO_UNITARIO_FAIXAS,Barato,Preço Médio,Caro,Total
QTD_UNIDADES_VENDIDAS_MES_FAIXAS,,,,
Nenhum,0.0,12.0,4.0,7.0
Baixa,40.0,33.0,56.0,45.0
Média,20.0,29.0,22.0,25.0
Alta,40.0,12.0,19.0,18.0
Muito Alta,0.0,12.0,0.0,5.0


In [16]:
# Isolando a seção "Informatica"
database_questao3_filter = database_questao3[database_questao3['SECAO_PRODUTO'] == 'Informática']
pd.crosstab(database_questao3_filter['QTD_UNIDADES_VENDIDAS_MES_FAIXAS'], database_questao3_filter['PRECO_UNITARIO_FAIXAS'], normalize = 'columns', margins=True, margins_name='Total').round(2) * 100

PRECO_UNITARIO_FAIXAS,Preço Médio,Caro,Total
QTD_UNIDADES_VENDIDAS_MES_FAIXAS,,,
Nenhum,0.0,8.0,5.0
Baixa,11.0,51.0,39.0
Média,6.0,18.0,14.0
Alta,22.0,15.0,18.0
Muito Alta,61.0,8.0,25.0


In [17]:
# Isolando a seção "Papelaria"
database_questao3_filter = database_questao3[database_questao3['SECAO_PRODUTO'] == 'Papelaria']
pd.crosstab(database_questao3_filter['QTD_UNIDADES_VENDIDAS_MES_FAIXAS'], database_questao3_filter['PRECO_UNITARIO_FAIXAS'], normalize = 'columns', margins=True, margins_name='Total').round(2) * 100

PRECO_UNITARIO_FAIXAS,Muito Barato,Barato,Preço Médio,Caro,Total
QTD_UNIDADES_VENDIDAS_MES_FAIXAS,,,,,
Nenhum,0.0,6.0,3.0,0.0,3.0
Baixa,7.0,22.0,29.0,50.0,18.0
Média,11.0,16.0,37.0,50.0,19.0
Alta,32.0,34.0,23.0,0.0,31.0
Muito Alta,51.0,21.0,9.0,0.0,31.0


Utilizando a segmentação por quartis para as variáveis PRECO_UNITARIO e QTD_UNIDADES_VENDIDAS_MES, foi possível observar — por meio de tabelas de frequência de dupla entrada — uma relação inversamente proporcional entre o preço e o volume de vendas.

Analisando a tabela geral de forma mais detalhada, nota-se que a faixa de preço classificada como "Muito Barato" está fortemente concentrada nas categorias de quantidade "Alta" e "Muito Alta", indicando que produtos com preços mais acessíveis tendem a apresentar alta rotatividade. Esse comportamento é valioso para o planejamento de estoque, pois sugere que esses produtos devem ter reposição constante.

Por outro lado, os produtos na faixa de preço "Caro" estão majoritariamente associados às faixas de quantidade "Baixa" e "Média", reforçando o comportamento inverso — produtos mais caros tendem a ter uma saída mais lenta.

Ao segmentar essa análise por seção, temos:

* Acessórios:
Esta seção não apresenta produtos na faixa de preço "Muito Barato". Os produtos classificados como "Barato" mostraram uma distribuição de vendas que varia entre "Baixa" e "Alta". Ainda assim, o padrão geral é mantido: produtos com preços mais altos tendem a ter um volume de vendas menor.

* Informática:
Esta seção exibe uma menor variedade de preços, com a maioria dos produtos concentrada nas faixas "Preço Médio" e "Caro". Dentro da faixa de menor preço para a seção ("Preço Médio"), os produtos estão concentrados na categoria "Muito Alta" de vendas, enquanto os produtos mais caros consistentemente apresentam baixa quantidade vendida. Esse contraste reforça a tendência inversa entre preço e volume de forma acentuada nesta seção.

* Papelaria:
É a única seção que abrange todas as quatro faixas de preço ('Muito Barato', 'Barato', 'Preço Médio', 'Caro'). O padrão observado segue claramente o comportamento geral da loja: produtos "Muito Baratos" possuem frequência de vendas muito alta, enquanto os produtos mais caros aparecem predominantemente nas faixas de quantidade "Baixa" e "Média".

### **QUESTÃO 4**

#### Considerando os produtos com maior faturamento bruto, quantos deles foram responsáveis por 80% do faturamento bruto total da loja?

In [18]:
database_questao4 = database.copy()
database_questao4["FATURAMENTO_BRUTO"] = database_questao4["PRECO_UNITARIO"] * database_questao4["QTD_UNIDADES_VENDIDAS_MES"]
database_questao4_ordenada = database_questao4.sort_values(by="FATURAMENTO_BRUTO", ascending=False)

# Calcular faturamento acumulado, arredondando por 3 casas
database_questao4_ordenada["FATURAMENTO_BRUTO_ACUMULADO"] = database_questao4_ordenada["FATURAMENTO_BRUTO"].cumsum()
faturamento_total = database_questao4_ordenada["FATURAMENTO_BRUTO"].sum()
database_questao4_ordenada["FATURAMENTO_PCT_ACUMULADO"] = (database_questao4_ordenada["FATURAMENTO_BRUTO_ACUMULADO"] / faturamento_total).round(3)

# Contar quantos produtos acumulam até 80% do faturamento
produtos_80_fat = database_questao4_ordenada[(database_questao4_ordenada["FATURAMENTO_PCT_ACUMULADO"] <= 0.80)]

qtd_faturamento80 = len(produtos_80_fat)
qtd_total = len(database_questao4)

# quantidade da base filtrada
print(f"A base filtrada possui {qtd_faturamento80} produtos")
print(f"A base total possui {qtd_total} produtos")

porcent_produtos = round((qtd_faturamento80 / qtd_total),3)  * 100

print(f"Isso representa {porcent_produtos}% dos produtos")

A base filtrada possui 49 produtos
A base total possui 290 produtos
Isso representa 16.900000000000002% dos produtos


Considerando o faturamento bruto individual dos produtos (resultado do PRECO_UNITARIO multiplicado pela QTD_UNIDADES_VENDIDAS_MES) e, em seguida, calculando o faturamento acumulado e seu percentual em relação ao total da loja (com arredondamento para três casas decimais), foi identificado que 49 produtos são responsáveis por 80% do faturamento bruto total.

Dado que a base de dados completa possui 290 produtos, esses 49 produtos representam aproximadamente 16,9% do portfólio total. Este achado evidencia um claro comportamento de Efeito Pareto (também conhecido como regra 80/20), onde uma parcela relativamente pequena de produtos gera a maior parte da receita bruta da loja.

### **QUESTÃO 5**

#### Quais são os 3 produtos com menor faturamento bruto em cada seção?

In [19]:
# Criação de base e dos campos com valor do faturamento bruto e %

database_questao5 = database.copy()
database_questao5['FATURAMENTO_BRUTO'] = database_questao5['PRECO_UNITARIO'] * database_questao5['QTD_UNIDADES_VENDIDAS_MES']

# Mostrar os principais produtos por secao e categorias
database_questao5_agrup = database_questao5.groupby(['SECAO_PRODUTO','CATEGORIA_PRODUTO' ,'COD_PRODUTO']).agg(
        PRECO_UNITARIO_MEDIA=('PRECO_UNITARIO', 'mean'),
        QTD_UNIDADES_VENDIDAS_MES_SUM=('QTD_UNIDADES_VENDIDAS_MES', 'sum'),
        FATURAMENTO_BRUTO_SUM=('FATURAMENTO_BRUTO', 'sum')
).round(2).sort_values(by=["SECAO_PRODUTO", "FATURAMENTO_BRUTO_SUM"], ascending=[True, True])

In [20]:
# Base considerando os produtos não tiveram vendas
database_questao5_agrup_fat0 =  database_questao5_agrup[database_questao5_agrup['FATURAMENTO_BRUTO_SUM'] == 0]
database_questao5_agrup_fat0

PRECO_UNITARIO_MEDIA  \
SECAO_PRODUTO CATEGORIA_PRODUTO      COD_PRODUTO                          
Acessórios    Calculadora Científica COD_P_321815                  89.9   
              Fone de Ouvido sem Fio COD_P_244216                 139.8   
              Tripé                  COD_P_173986                  63.0   
                                     COD_P_315008                  47.9   
Informática   Impressora a Jato      COD_P_324413                 629.9   
              Notebook               COD_P_211988                4399.9   
              Notebook Premium       COD_P_316388                4999.9   
Papelaria     Agenda                 COD_P_316501                  65.9   
              Cola e Fita Adesiva    COD_P_153079                   5.8   
              Estilete               COD_P_303835                   8.9   
              Marca Texto            COD_P_136800                   6.9   
              Tinta e Pincel         COD_P_318311                   8.9   

                                                   QTD_UNIDADES_VENDIDAS_MES_SUM  \
SECAO_PRODUTO CATEGORIA_PRODUTO      COD_PRODUTO                                   
Acessórios    Calculadora Científica COD_P_321815                              0   
              Fone de Ouvido sem Fio COD_P_244216                              0   
              Tripé                  COD_P_173986                              0   
                                     COD_P_315008                              0   
Informática   Impressora a Jato      COD_P_324413                              0   
              Notebook               COD_P_211988                              0   
              Notebook Premium       COD_P_316388                              0   
Papelaria     Agenda                 COD_P_316501                              0   
              Cola e Fita Adesiva    COD_P_153079                              0   
              Estilete               COD_P_303835                              0   
              Marca Texto            COD_P_136800                              0   
              Tinta e Pincel         COD_P_318311                              0   

                                                   FATURAMENTO_BRUTO_SUM  
SECAO_PRODUTO CATEGORIA_PRODUTO      COD_PRODUTO                          
Acessórios    Calculadora Científica COD_P_321815                    0.0  
              Fone de Ouvido sem Fio COD_P_244216                    0.0  
              Tripé                  COD_P_173986                    0.0  
                                     COD_P_315008                    0.0  
Informática   Impressora a Jato      COD_P_324413                    0.0  
              Notebook               COD_P_211988                    0.0  
              Notebook Premium       COD_P_316388                    0.0  
Papelaria     Agenda                 COD_P_316501                    0.0  
              Cola e Fita Adesiva    COD_P_153079                    0.0  
              Estilete               COD_P_303835                    0.0  
              Marca Texto            COD_P_136800                    0.0  
              Tinta e Pincel         COD_P_318311                    0.0

In [21]:
# Base considerando os produtos que tiveram faturamento
database_questao5_filter =  database_questao5_agrup[database_questao5_agrup['FATURAMENTO_BRUTO_SUM'] > 0]
database_questao5_filter = database_questao5_filter.groupby('SECAO_PRODUTO').head(3)
database_questao5_filter

PRECO_UNITARIO_MEDIA  \
SECAO_PRODUTO CATEGORIA_PRODUTO  COD_PRODUTO                          
Acessórios    Cabo USB           COD_P_228204                 17.90   
              Suporte de Celular COD_P_332956                 15.00   
                                 COD_P_182923                 24.90   
Informática   Pen Drive          COD_P_181540                 42.90   
              Mouse Sem Fio      COD_P_236215                105.90   
              Mouse Óptico       COD_P_242767                 52.90   
Papelaria     Brinde             COD_B_000002                  0.01   
                                 COD_B_000001                  0.01   
                                 COD_B_000004                  0.01   

                                               QTD_UNIDADES_VENDIDAS_MES_SUM  \
SECAO_PRODUTO CATEGORIA_PRODUTO  COD_PRODUTO                                   
Acessórios    Cabo USB           COD_P_228204                              3   
              Suporte de Celular COD_P_332956                              4   
                                 COD_P_182923                              3   
Informática   Pen Drive          COD_P_181540                              1   
              Mouse Sem Fio      COD_P_236215                              2   
              Mouse Óptico       COD_P_242767                              5   
Papelaria     Brinde             COD_B_000002                             49   
                                 COD_B_000001                             62   
                                 COD_B_000004                             98   

                                               FATURAMENTO_BRUTO_SUM  
SECAO_PRODUTO CATEGORIA_PRODUTO  COD_PRODUTO                          
Acessórios    Cabo USB           COD_P_228204                  53.70  
              Suporte de Celular COD_P_332956                  60.00  
                                 COD_P_182923                  74.70  
Informática   Pen Drive          COD_P_181540                  42.90  
              Mouse Sem Fio      COD_P_236215                 211.80  
              Mouse Óptico       COD_P_242767                 264.50  
Papelaria     Brinde             COD_B_000002                   0.49  
                                 COD_B_000001                   0.62  
                                 COD_B_000004                   0.98

Ao analisar o faturamento bruto (calculado como PRECO_UNITARIO multiplicado pela QTD_UNIDADES_VENDIDAS_MES), foi observado que alguns produtos registraram faturamento bruto igual a zero, indicando que não tiveram vendas no último mês. Para os produtos que apresentaram faturamento positivo, identificamos os 3 menores em cada seção:

Produtos com Faturamento Bruto de R$ 0,00 (Sem Vendas no Mês):

* Acessórios:

(COD_P_321815, Calculadora Científica)
(COD_P_244216, Fone de Ouvido sem Fio)
(COD_P_173986, Tripé)
(COD_P_315008, Tripé)

* Informática:

(COD_P_324413, Impressora a Jato)
(COD_P_211988, Notebook)
(COD_P_316388, Notebook Premium)

* Papelaria:

(COD_P_316501, Agenda)
(COD_P_153079, Cola e Fita Adesiva)
(COD_P_303835, Estilete)
(COD_P_136800, Marca Texto)
(COD_P_318311, Tinta e Pincel)

Para os 3 produtos com o menor Faturamento Bruto (acima de R$ 0,00) em cada seção:

* Acessórios:

(COD_P_228204, Cabo USB)
(COD_P_332956, Suporte de Celular)
(COD_P_182923, Suporte de Celular)

* Informática:

(COD_P_181540, Pen Drive)
(COD_P_236215, Mouse Sem Fio)
(COD_P_242767, Mouse Óptico)

* Papelaria:

(COD_B_000002, Brinde)
(COD_B_000001, Brinde)
(COD_B_000004, Brinde)

Essa distinção entre produtos sem vendas e produtos com as menores vendas positivas é crucial para a gestão de estoque e o planejamento estratégico. Os produtos com faturamento zero podem indicar obsolescência, problemas de marketing ou precificação, enquanto aqueles com faturamento baixo, mas positivo, ainda podem ter alguma demanda. Os "Brindes" na seção de Papelaria, por exemplo, como esperado, apresentam o menor faturamento devido ao seu preço simbólico.

### **QUESTÃO 6**

#### Existem produtos com valores atípicos de quantidade de unidades vendidas? E de preço unitário? E de faturamento bruto?

In [22]:
database_questao6 = database.copy()

# Selecionando as variáveis
unidades_vendidas = database_questao6['QTD_UNIDADES_VENDIDAS_MES']
preco_unitario = database_questao6['PRECO_UNITARIO']
faturamento_bruto = database_questao6['PRECO_UNITARIO'] * database_questao6['QTD_UNIDADES_VENDIDAS_MES']

# Criando método para imprimir os dados comum para análise das variáveis
def detalhes_variavel(var):
    print("O mínimo é", var.min())
    print("O percentil 1 é", var.quantile(0.01).round(1))
    print("O máximo é", var.max())
    print("O percentil 99 é", var.quantile(0.99).round(1))
    print("A amplitude é", var.max() - var.min())
    print("A amplitude percentílica é", (var.quantile(0.99) - var.quantile(0.01)).round(1))
    print("A média é", var.mean())
    print("A mediana é", var.median())

    limite_inferior = var.quantile(0.25) - 1.5 * (var.quantile(0.75) - var.quantile(0.25))
    limite_superior = var.quantile(0.75) + 1.5 * (var.quantile(0.75) - var.quantile(0.25))

    print(f"Limite Inferior: {limite_inferior}")
    print(f"Limite Superior: {limite_superior}")
    print(f"Regra Tradicional do Intervalo Interquartil Inferior: {var[var < limite_inferior].count()}")
    print(f"Regra Tradicional do Intervalo Interquartil Superior: {var[var > limite_superior].count()}")


In [23]:
# Observar 20 menores da variavel QTD_UNIDADES_VENDIDAS_MES
unidades_vendidas.nsmallest(20)

,QTD_UNIDADES_VENDIDAS_MES
53,0
69,0
98,0
112,0
124,0
126,0
144,0
153,0
162,0
194,0


In [24]:
# Observar 20 maiores da variavel QTD_UNIDADES_VENDIDAS_MES
unidades_vendidas.nlargest(20)

,QTD_UNIDADES_VENDIDAS_MES
80,132
268,98
164,79
149,74
147,62
186,49
185,41
45,40
270,40
57,39


In [25]:
# imprimir os detalhes da variável
detalhes_variavel(unidades_vendidas)

O mínimo é 0
O percentil 1 é 0.0
O máximo é 132
O percentil 99 é 74.6
A amplitude é 132
A amplitude percentílica é 74.6
A média é 12.124137931034483
A mediana é 7.5
Limite Inferior: -14.0
Limite Superior: 34.0
Regra Tradicional do Intervalo Interquartil Inferior: 0
Regra Tradicional do Intervalo Interquartil Superior: 13


In [26]:
# Observar 20 menores da variavel PRECO_UNITARIO
preco_unitario.nsmallest(20)

,PRECO_UNITARIO
80,0.01
147,0.01
186,0.01
268,0.01
85,0.50
42,0.60
76,0.60
242,0.70
36,0.80
164,0.80


In [27]:
# Observar 20 maiores da variavel PRECO_UNITARIO
preco_unitario.nlargest(20)

,PRECO_UNITARIO
241,5850.0
144,4999.9
179,4800.0
98,4399.9
278,4399.9
284,4299.9
74,3699.9
11,3299.9
249,3199.9
54,2980.0


In [28]:
# imprimir os detalhes da variável
detalhes_variavel(preco_unitario)

O mínimo é 0.01
O percentil 1 é 0.0
O máximo é 5850.0
O percentil 99 é 4443.9
A amplitude é 5849.99
A amplitude percentílica é 4443.9
A média é 233.6653103448276
A mediana é 18.4
Limite Inferior: -122.53750000000001
Limite Superior: 217.3625
Regra Tradicional do Intervalo Interquartil Inferior: 0
Regra Tradicional do Intervalo Interquartil Superior: 37


In [29]:
# Observar 20 menores da variavel faturamento_bruto
faturamento_bruto.nsmallest(20)

,0
53,0.00
69,0.00
98,0.00
112,0.00
124,0.00
126,0.00
144,0.00
153,0.00
162,0.00
194,0.00


In [30]:
# Observar 20 maiores da variavel faturamento_bruto
faturamento_bruto.nlargest(20)

,0
54,17880.0
278,17599.6
66,9596.0
11,6599.8
249,6399.8
265,5998.8
241,5850.0
109,5699.7
181,4899.3
179,4800.0


In [31]:
# imprimir os detalhes da variável
detalhes_variavel(faturamento_bruto)

O mínimo é 0.0
O percentil 1 é 0.0
O máximo é 17880.0
O percentil 99 é 6929.4
A amplitude é 17880.0
A amplitude percentílica é 6929.4
A média é 711.4448620689656
A mediana é 121.94999999999999
Limite Inferior: -609.7375000000001
Limite Superior: 1132.5625
Regra Tradicional do Intervalo Interquartil Inferior: 0
Regra Tradicional do Intervalo Interquartil Superior: 45


* Unidades Vendidas

Analisando os 20 menores valores, observa-se apenas um intervalo muito estreito (entre 0 e 1). Já entre os maiores valores, há uma maior dispersão — com destaques para quantidades como 132, 98, 79, 74 e 62.

Aplicando o método do intervalo interquartil (IQR):

Não há outliers no limite inferior.
Pelo limite superior, qualquer valor acima de 34 unidades vendidas é considerado outlier, o que inclui 13 produtos.

Outro ponto relevante é a grande diferença entre o valor máximo e o percentil 99, o que reforça a presença de valores atípicos superiores.

* Preço Unitário

Entre os 20 menores preços, destacam-se quatro valores de R$ 0,01, que possivelmente representam brindes.

Nos maiores preços, há uma discrepância significativa:

O maior valor (R\$ 5.850,00) é aproximadamente 9,5 vezes maior que o 20º maior valor (R\$ 609,90).
Saltos expressivos começam a partir de R\$ 1.560,00, com acréscimos de quase R\$ 300 entre valores consecutivos.

Pelo IQR:

Nenhum outlier pelo limite inferior.
Pelo limite superior, valores acima de R\$ 217,36 são considerados outliers, resultando em 37 produtos fora do intervalo esperado.

Além disso, nota-se um descolamento considerável entre média e mediana, o que indica que valores extremos estão puxando a média para cima.

* Faturamento Bruto

Nos 20 menores faturamentos, há pouca variação entre os valores. Por outro lado, entre os maiores, há discrepâncias marcantes:
Os dois maiores faturamentos (R\$ 17.880,00 e R\$ 17.599,60) são quase o dobro do terceiro maior valor.
O terceiro valor, por sua vez, também apresenta um salto significativo em relação ao quarto.

Segundo o IQR:

Não há outliers inferiores.
Valores acima de R$ 1.132,56 são considerados outliers, totalizando 45 produtos acima do limite.

Por fim, assim como nos preços unitários, há forte assimetria entre média e mediana, além de uma diferença relevante entre amplitude total e amplitude percentílica, o que reforça a presença de valores atípicos superiores.

### **QUESTÃO 7**

#### Qual seção de produtos tem maior variabilidade de preços unitários?

In [32]:
database_questao7 = database.copy()

# Selecionando variáveis
preco = database_questao7['PRECO_UNITARIO']
secao = database_questao7['SECAO_PRODUTO']

# Criação do dataframe com o describe do preço por seção
database_questao7_agrup = pd.DataFrame(preco.groupby(secao).describe())

# Criação das colunas com os cálculos para comparação
database_questao7_agrup['AMPLITUDE'] = database_questao7_agrup['max'] - database_questao7_agrup['min']
p01 = preco.groupby(secao).quantile(0.01)
p99 = preco.groupby(secao).quantile(0.99)
database_questao7_agrup['AMPLITUDE_PERCENTILICA'] = (p99 - p01).round(1)
desvio_abs_medio = preco.groupby(secao).apply(lambda x: np.mean(np.abs(x - x.mean())))
database_questao7_agrup['DESVIO_ABSOLUTO_MEDIO'] = desvio_abs_medio
desvio_abs_mediano = preco.groupby(secao).apply(lambda x: np.median(np.abs(x - np.median(x))))
database_questao7_agrup['DESVIO_ABSOLUTO_MEDIANO'] = desvio_abs_mediano
database_questao7_agrup['COEFICIENTE_VARIACAO'] = (database_questao7_agrup['std'] / database_questao7_agrup['mean']).round(3)  * 100
database_questao7_agrup['COEFICIENTE_VARIACAO_MEDIANO'] = (database_questao7_agrup['std'] / database_questao7_agrup['50%']).round(3)  * 100

# Exibir base
database_questao7_agrup


,count,mean,std,min,25%,50%,75%,max,AMPLITUDE,AMPLITUDE_PERCENTILICA,DESVIO_ABSOLUTO_MEDIO,DESVIO_ABSOLUTO_MEDIANO,COEFICIENTE_VARIACAO,COEFICIENTE_VARIACAO_MEDIANO
SECAO_PRODUTO,,,,,,,,,,,,,,
Acessórios,56.0,103.255357,77.600337,12.90,37.925,89.9,139.825,294.9,282.00,273.8,62.677551,50.05,75.2,86.3
Informática,57.0,1047.996491,1580.448485,31.90,82.900,238.9,1049.000,5850.0,5818.10,5340.3,1211.128163,186.00,150.8,661.6
Papelaria,177.0,12.682712,18.714275,0.01,2.500,6.9,13.900,129.9,129.89,92.3,11.488668,4.90,147.6,271.2


A análise da variabilidade dos preços unitários por seção, utilizando o Coeficiente de Variação (CV), revela que a seção Informática possui a maior variabilidade.

Coeficiente de Variação (CV) baseado na Média:<br>
Informática: 150,8%<br>
Papelaria: 147,6%<br>
Acessórios: 75,2%<br>
Inicialmente, com base no CV padrão (desvio padrão dividido pela média), as seções de Informática e Papelaria apresentam variabilidades próximas e elevadas, enquanto Acessórios mostra uma distribuição de preços mais homogênea.

No entanto, ao aprofundar a análise com o Coeficiente de Variação Mediano (desvio padrão dividido pela mediana, capturando melhor a assimetria e o impacto de valores extremos), a diferença se torna mais expressiva:

O CV Mediano de Informática aumenta drasticamente de 150,8% para 662,6%.<br>
O CV Mediano de Papelaria sobe de 147,6% para 271,2%.<br>
Essa discrepância acentuada entre o CV tradicional e o CV Mediano, particularmente na seção de Informática, indica uma maior assimetria e um impacto mais forte de valores extremos (outliers) nos preços unitários. Isso significa que, embora a Papelaria também seja heterogênea, a seção de Informática apresenta produtos com preços que se desviam muito mais da "normalidade" do seu centro (mediana), o que reforça sua posição como a seção com a maior variabilidade de preços unitários.

### **QUESTÃO 8**

#### Existem produtos com alta quantidade de unidades vendidas e baixo preço? Existem produtos com baixa quantidade de unidades vendidas e alto preço?

In [33]:
database_questao8 = database.copy()

In [34]:
preco_baixo = database_questao8['PRECO_UNITARIO'].quantile(0.25)
alta_quantidade_vendidas = database_questao8['QTD_UNIDADES_VENDIDAS_MES'].quantile(0.75)

print(preco_baixo)
print(alta_quantidade_vendidas)

# base com filtro de menor preço x alta quantidade de venda
database_questao8_baixo_preco = database_questao8[
    (database_questao8['PRECO_UNITARIO'] < preco_baixo) &
    (database_questao8['QTD_UNIDADES_VENDIDAS_MES'] > alta_quantidade_vendidas)
    ]
database_questao8_baixo_preco.sort_values(by=["QTD_UNIDADES_VENDIDAS_MES"],ascending=False)

4.925000000000001
16.0


,COD_PRODUTO,CATEGORIA_PRODUTO,SECAO_PRODUTO,PRECO_UNITARIO,QTD_UNIDADES_VENDIDAS_MES
80,COD_B_000003,Brinde,Papelaria,0.01,132
268,COD_B_000004,Brinde,Papelaria,0.01,98
164,COD_P_291073,Envelope,Papelaria,0.80,79
149,COD_P_141165,Lápis,Papelaria,1.20,74
147,COD_B_000001,Brinde,Papelaria,0.01,62
186,COD_B_000002,Brinde,Papelaria,0.01,49
185,COD_P_243744,Caneta Simples,Papelaria,1.90,41
45,COD_P_304471,Caneta Simples,Papelaria,1.90,40
57,COD_P_187818,Envelope,Papelaria,1.00,39
169,COD_P_126057,Lápis,Papelaria,1.50,36


In [35]:
preco_alto = database_questao8['PRECO_UNITARIO'].quantile(0.75)
baixa_quantidade_vendidas = database_questao8['QTD_UNIDADES_VENDIDAS_MES'].quantile(0.25)

print(preco_alto)
print(baixa_quantidade_vendidas)

# base com filtro de maior preço x baixa quantidade de venda
database_questao8_alto_preco = database_questao8[
    (database_questao8['PRECO_UNITARIO'] > preco_alto) &
    (database_questao8['QTD_UNIDADES_VENDIDAS_MES'] < baixa_quantidade_vendidas)
    ]
database_questao8_alto_preco.sort_values(by=["QTD_UNIDADES_VENDIDAS_MES"],ascending=False)

89.9
4.0


,COD_PRODUTO,CATEGORIA_PRODUTO,SECAO_PRODUTO,PRECO_UNITARIO,QTD_UNIDADES_VENDIDAS_MES
2,COD_P_178413,Monitor,Informática,799.9,3
3,COD_P_316368,Mochila Escolar,Acessórios,139.0,3
109,COD_P_129420,Tablet,Informática,1899.9,3
31,COD_P_231063,Teclado Mecânico,Informática,299.9,3
167,COD_P_162491,HD Externo,Informática,238.9,3
276,COD_P_298708,Mochila Executiva,Acessórios,239.9,3
90,COD_P_234684,Monitor,Informática,1049.0,3
115,COD_P_187735,Mochila Executiva,Acessórios,279.9,3
145,COD_P_173551,Carregador Portátil,Acessórios,128.9,3
11,COD_P_280958,Notebook,Informática,3299.9,2


Para esta análise, foi utilizado os quartis como critério: valores altos foram definidos como aqueles acima do terceiro quartil (Q3), e valores baixos como aqueles abaixo do primeiro quartil (Q1) para ambas as variáveis (QTD_UNIDADES_VENDIDAS_MES e PRECO_UNITARIO). Com base nisso, os resultados são os seguintes:

Produtos com Alta Quantidade de Unidades Vendidas (> 16 unidades) e Baixo Preço (&lt; R$ 4,93):<br>
Foram identificados 37 produtos que se enquadram neste perfil. Todos esses produtos pertencem à seção Papelaria. Ao observar suas categorias, percebe-se que são predominantemente itens de uso recorrente e baixo valor unitário, como canetas, lápis, envelopes e borrachas. Esses produtos tendem a apresentar alta rotatividade, o que reforça sua importância estratégica para a reposição constante de estoque.

Produtos com Baixa Quantidade de Unidades Vendidas (&lt; 4 unidades) e Alto Preço (> R$ 89,90):<br>
Neste grupo, foram identificados 36 produtos, que estão distribuídos entre todas as seções (Acessórios, Informática e Papelaria). Ao contrário do grupo anterior, esses itens possuem alto valor unitário e são geralmente produtos com maior durabilidade ou um ciclo de reposição mais longo, como monitores, notebooks, tablets e mochilas executivas.